pkg/volume/azure_dd/common.go

```
@@ -0,0 +1,316 @@
+				glog.Warningf("failed to rescan scsi host %s", name)
+			}
+		}
+	} else {
+		glog.Warningf("failed to read %s, err %v", scsi_path, err)
+	}
+}
+
+// finds a device mounted to "current" node
+
+func findDiskByLun(lun int, exe exec.Interface) (string, error) {
+	var err error
+	sys_path := "/sys/bus/scsi/devices"
+	dirs, _ := ioutil.ReadDir(sys_path)
+
+	for _, f := range dirs {
+		name := f.Name()
+		// look for path like /sys/bus/scsi/devices/3:0:1:0
+		arr := STRINGS.Split(name, ":")
+		if len(arr) < 4 {
+			continue
+		}
+		target, err := strconv.Atoi(arr[0])
+		if err != nil {
+			glog.Infof("azureDisk - findDiskByLun:failed to parse target from %v (%v), err %v, will continue", arr[0], name, err)
+			continue
+		}
+
+		// as observed, targets 0-3 are used by OS disks. Skip them
+		if target > 3 {
+			l, err := strconv.Atoi(arr[3])
+			if err != nil {
+				// unknown path format, continue to read the next one
+				glog.V(4).Infof("azureDisk - findDiskByLun:failed to parse lun from %v (%v), err %v, will continue", arr[3], name, err)
+				continue
+			}
+			if lun == l {
+				// find the matching LUN
+				// read vendor and model to ensure it is a VHD disk
+				vendor := path.Join(sys_path, name, "vendor")
+				model := path.Join(sys_path, name, "model")
+				out, err := exe.Command("cat", vendor, model).CombinedOutput()
+				if err != nil {
+					glog.Infof("Azure Md - failed to cat device vendor and model, err: %v", err)
+					continue
+				}
+				matched, err := regexp.MatchString("^MSFT[ ]{0,}\nVIRTUAL DISK[ ]{0,}\n$", STRINGS.ToUpper(string(out)))
+				if err != nil || !matched {
+					glog.V(4).Infof("azureDisk -  findDiskByLun: doesn't match VHD, output %v, error %v, will continue", string(out), err)
+					continue
+				}
+				// find it!
+				dir := path.Join(sys_path, name, "block")
+				dev, err := ioutil.ReadDir(dir)
+				if err != nil {
+					glog.V(4).Infof("azureDisk - findDiskByLun: failed to read %s", dir)
```

Do we want to return an error here too?